provider.tf

```diff
@@ -23,6 +28,11 @@ provider "aws" {
       var.common_tags
     )
   }
+  assume_role {
+    role_arn     = var.assume_role_arn != "" ? var.assume_role_arn : null
+    session_name = var.assume_role_arn != "" ? (var.assume_role_session_name != "" ? var.assume_role_session_name : null) : null # Optional: for better tracking; Terraform will generate one if omitted
```

Comment:
Alternatively, we could put some validations for these, I think. For example, if `assume_role_session_name`, then `assume_role_arn` must be set as well.